### PR TITLE
deprecate router-id option from ospf_interfaces

### PR DESCRIPTION
**Pull request by rohitthakur2590**
##### SUMMARY

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1236
resolve: 
[x] https://github.com/ansible-collections/junipernetworks.junos/issues/215
[x] https://github.com/ansible-collections/junipernetworks.junos/issues/198

##### ISSUE TYPE

- Bugfix Pull Request
 

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: tests/integration/targets/junos_ospf_interfaces/tests/netconf/parsed.yaml [hunk 1]
```diff
@@ -13,7 +13,6 @@
               metric: 5
               priority: 3
```
Comment (NilashishC) [inline tests/integration/targets/junos_ospf_interfaces/tests/netconf/parsed.yaml:14]: @rohitthakur2590 can we keep a test for this option to make sure that it works until it is removed?

File: changelogs/fragments/junos_ospf_router_if_fix.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+deprecated_features:
+  - Deprecated router_id from ospf-interfaces resource module.
```
Comment (NilashishC) [inline changelogs/fragments/junos_ospf_router_if_fix.yaml:3]: We are deprecating it right now, but it does not have an alternative (i.e., junos_routing_options). I think we should deprecate it only when a correct option is available. To get past the reported traceback, maybe we just keep the if-condition.

**Review by NilashishC — CHANGES_REQUESTED** [review]

**Review by NilashishC — APPROVED** [review]

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Pull request merged**